### ISSUE TITLE
Docker image pytorch/pytorch:2.1.2-cuda12.1-cudnn8-runtime uses python 3.10, but has libboost_python38 installed

### ISSUE THREAD

**Issue by jxtps** [pytorch/pytorch#119843]
### 🐛 Describe the bug

I'm trying to use the [`pytorch/pytorch:2.1.2-cuda12.1-cudnn8-runtime`](https://hub.docker.com/layers/pytorch/pytorch/2.1.2-cuda12.1-cudnn8-runtime/images/sha256-3387e598cb94fc248d82e712a65b10931a990cea3a2e76362ca30d135f565de4?context=explore) docker image (listed on https://hub.docker.com/r/pytorch/pytorch/tags) and need to use a python package that requires libboost_python + compilation from scratch (no wheels provided :-/ ). 

Unfortunately, when I install the package the installation fails with: 
```
 /opt/conda/compiler_compat/ld: cannot find -lboost_python: No such file or directory
```

libboost-python-dev is indeed installed in that docker image, but: 
```
# ls -l /usr/lib/x86_64-linux-gnu/libboost_python*
-rw-r--r-- 1 root root 701312 Apr  1  2020 /usr/lib/x86_64-linux-gnu/libboost_python38.a
lrwxrwxrwx 1 root root     27 Apr  1  2020 /usr/lib/x86_64-linux-gnu/libboost_python38.so -> libboost_python38.so.1.71.0
-rw-r--r-- 1 root root 259776 Apr  1  2020 /usr/lib/x86_64-linux-gnu/libboost_python38.so.1.71.0
```
i.e. it appears to only be built for python 3.8 (presumably the python version that was used in earlier versions of the docker image?)

However, the "primary" python version in the docker image is now: 
```
# python --version
Python 3.10.13
```
Please make the libboost_python target the "primary" python version in the docker image. Thank you!

### Versions

[pytorch/pytorch:2.1.2-cuda12.1-cudnn8-runtime](https://hub.docker.com/layers/pytorch/pytorch/2.1.2-cuda12.1-cudnn8-runtime/images/sha256-3387e598cb94fc248d82e712a65b10931a990cea3a2e76362ca30d135f565de4?context=explore)

**Comment by jxtps** [comment]
Hmm... I just realized that it's my Dockerfile that does the installation of `libboost-python-dev`, and that it's actually not included in `pytorch/pytorch:2.1.2-cuda12.1-cudnn8-runtime`.

**Comment by jxtps** [comment]
The docker image uses ubuntu 20.04, for which the default python appears to be 3.8, so all packages that apt has access to are built for python 3.8. 

But pytorch is only installed for python 3.10. 

This makes it exceedingly frustrating to work with it. 

**Please use consistent versions of ubuntu & python for the docker images**
